app/services/CheckoutService.scala

```diff
@@ -214,8 +218,8 @@ class CheckoutService(
 
     val personalData = subscribeRequest.genericData.personalData
 
-    // Set correspondence to delivery address iff recipient is not a giftee
-    val correspondenceAddress = subscribeRequest.productData.left.toOption.map(_.deliveryRecipient).filterNot(_.isGiftee).map(_.address)
+    // Set correspondence address to the buyer's mailing address if they have provided that data
```

Comment:
Is it appropriate to do this, even if they haven't opted into marketing etc.?

Reply:
The opt in is for emails. Any direct mail activity is based on legitimate interest and is used for other use-cases than standard marketing, and has an opt-out in the profile section.